### PR TITLE
Server container

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Add containerized server initial support. Some of the copied code is not yet fully functionnal, but it's enough for the testsuite for now

### CODE REVIEW

File: backend_modules/libvirt/host/main.tf [hunk 1]
```diff
@@ -13,7 +13,9 @@ locals {
     },
     contains(local.x86_64_v2_images, var.image) ? { cpu_model = "host-model", xslt = file("${path.module}/cpu_features.xsl") } : {},
     contains(var.roles, "server") ? { memory = 4096, vcpu = 2 } : {},
+    contains(var.roles, "server_containerized") ? { memory = 4096, vcpu = 2 } : {},
```
Comment (srbarrios) [inline backend_modules/libvirt/host/main.tf:16]: Keep in mind that the default memory value is not the value we use in most of our terraform files:
```
vcpu = 8
memory = 32768
```
I wonder if we should not change this default value for server and server_containerized.

File: salt/server_containerized/initial_content.sls [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+             desiredpassword={{ server_password }}&\
+             desiredpasswordConfirm={{ server_password }}&\
+             email=galaxy-noise%40suse.de&\
+             firstNames=Administrator&\
+             lastName=Administrator"
+    - verify_ssl: False
+    - unless: uyunictl exec "satwho | grep -x {{ server_username }} server_username"
+    - require:
+      - http: wait_for_tomcat
+      - sls: server_containerized.tools
```
Comment (srbarrios) [inline salt/server_containerized/initial_content.sls:45]: Wasn't that something we use to perform in the Core stage of Cucumber features?
Or that's something different?

**Review by srbarrios — APPROVED** [review]
Looks good to me.
Maybe in another iteration we can try to find code duplication between server and server_containerized and have a base_server from where both will extend.

File: backend_modules/libvirt/host/main.tf (hunk 1 above)
Comment (rjmateus) [reply backend_modules/libvirt/host/main.tf:16]: No, please. These values are defaults for most terraform users. Our CI is just one of the use cases, and it has the needed memory defined in the main.tf file that deploys it.
If a user is starting to use SUMAFORM it should not just get a 32Gb server out of the box.

Comment (srbarrios) [reply backend_modules/libvirt/host/main.tf:16]: Make sense. Still, keep this in mind so you set a proper value in the main.tf for our CI test environments using server_containerized. ;)

**Pull request merged**